mail_tracking/static/src/js/failed_message/widget.js

```diff
@@ -0,0 +1,311 @@
+/* Copyright 2019 Alexandre Díaz
+   License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html). */
+odoo.define('mail_tracking.FailedMessage', function (require) {
+    "use strict";
+
+    var AbstractField = require('web.AbstractField');
+    var BasicModel = require('web.BasicModel');
+    var BasicView = require('web.BasicView');
+    var Chatter = require('mail.Chatter');
+    var MailThread = require('mail.widget.Thread');
+    var utils = require('mail.utils');
+    var core = require('web.core');
+    var field_registry = require('web.field_registry');
+    var time = require('web.time');
+
+    var QWeb = core.qweb;
+
+
+    /**
+    * Helper method to fetch failed messages
+    *
+    * @private
+    * @param {Object} self
+    * @param {Array} ids
+    * @returns {Array}
+    */
+    function _readMessages (self, ids) {
```

Comment:
As this is a helper method, I think `self` is not very specific. How about this?:

```suggestion
    function _readMessages (widget, ids) {
```